### DISCUSSION TITLE
Error in Decode -  Cannot decode EPC with header C3

### DISCUSSION THREAD

**Discussion by mimansa-dev** [fulfilio/pyepc#9] [General]
I am trying to decode a EPC but get an error "pyepc.exceptions.DecodingError: Cannot decode EPC with header C3"

```
from pyepc import decode
sgtin = decode('C310821E1A27B82D49F00003')
```

The traceback is as follows. Appreciate if someone could guide me.

```
Traceback (most recent call last):
  File "/opt/app/virtual/cinetopia/lib64/python3.6/site-packages/pyepc/epc.py", line 183, in decode
    header_class_map[header]
KeyError: 'C3'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/app/virtual/cinetopia/lib64/python3.6/site-packages/pyepc/epc.py", line 186, in decode
    "Cannot decode EPC with header {}".format(header)
pyepc.exceptions.DecodingError: Cannot decode EPC with header C3

```

**Comment by sharoonthomas** [comment]
The package currently only implements headers 30 (SGTIN-98), 31 (SSCC-96) and 36 (SGTIN-198). 

3C prefix indicates CPI-96. 

Happy to review and merge a pull request with tests if you choose to implement. You can refer to existing implementations for above Decoders to implement.

More documentation here: https://www.gs1.org/sites/default/files/docs/epc/GS1_EPC_TDS_i1_10.pdf

**Reply by mimansa-dev** [reply]
Thanks for your response. However, I would like to seek the following clarifications.

I believe that 30 is the header value for SGTIN-96. Please confirm if there is something like SGTIN-98 also.

3C is indeed the header value for CPI-96 but in my case it is C3 not 3C that I am seeing in the EPC tag value. Does that mean that the tag is encoded incorrectly?

**Reply by sharoonthomas** [reply]
> I believe that 30 is the header value for SGTIN-96. Please confirm if there is something like SGTIN-98 also.

You are correct. Typo on my side. I meant 98.

> 3C is indeed the header value for CPI-96 but in my case it is C3 not 3C that I am seeing in the EPC tag value. Does that mean that the tag is encoded incorrectly?

From page 81 of the GS1 EPC standard, C3 is valid but says "Reserved until 64 bit sunset". So very likely that this is an incorrectly encoded EPC.

<img width="639" alt="image" src="https://user-images.githubusercontent.com/305807/194767074-661897b5-186f-481f-a022-e316f6e6b15d.png">
